File: backend/app/schemas/summaries.py
from datetime import date, datetime

from pydantic import BaseModel


class SummaryListItem(BaseModel):
    id: str
    period_label: str
    headline: str
    tags: list[str]
    word_count: int
    source_count: int
    generated_at: datetime
    status: str


class SummaryListResponse(BaseModel):
    items: list[SummaryListItem]
    total: int


class SummaryDetail(BaseModel):
    id: str
    period_label: str
    headline: str
    sections: dict[str, str]
    citations: dict[str, dict]
    tags: list[str]
    word_count: int
    source_count: int
    generated_at: datetime
    status: str
    reading_minutes: int

File: frontend/src/lib/summaries.ts
import { apiFetch } from "./api";

export type SummaryListItem = {
  id: string;
  period_label: string;
  headline: string;
  tags: string[];
  word_count: number;
  source_count: number;
  generated_at: string;
  status: string;
};

export type SummaryDetail = SummaryListItem & {
  sections: Record<string, string>;
  citations: Record<string, { label: string; value: string; source: string; indicator_id?: string }>;
  reading_minutes: number;
};

export async function fetchSummaries(region?: string) {
  const qs = region ? `?region=${region}` : "";
  return apiFetch<{ items: SummaryListItem[]; total: number }>(`/summaries${qs}`);
}

export async function fetchSummary(id: string) {
  return apiFetch<SummaryDetail>(`/summaries/${id}`);
}

export const SECTION_TITLES: Record<string, string> = {
  intro: "Введение",
  ru: "Россия",
  us: "США",
  eu: "Еврозона",
  fx: "Валюты",
  next_week: "На следующей неделе",
  risks: "Риски",
};
